stistools/defringe/normspflat.py

```diff
@@ -41,58 +52,309 @@ def normspflat(inflat, outflat, do_cal=True, biasfile=None, darkfile=None, pixel
+                output_suffix = 'sx2'  # geometrically-corrected
+            elif opt_elem == 'G750L':
+                output_suffix = 'crj'  # not geometrically-corrected
+                for keyword in OMIT_G750L:
+                    if f[0].header[keyword] == 'PERFORM':
+                        f[0].header[keyword] = 'OMIT'
+
+        # Call calstis:
+        old_cwd = os.getcwd()  # outflat and ref_vars are abs paths
+        try:
+            os.chdir(os.path.dirname(inflat))  # Must be in same directory to pick up EPC file
+            trailer = os.path.join(os.path.dirname(outflat),
+                                   '{}_calstis.log'.format(rootname))
+            outroot = os.path.dirname(outflat) + os.sep
+            outname = os.path.join(os.path.dirname(outflat),
+                                   '{}_{}.fits'.format(rootname, output_suffix))
+            # Remove files from previous iterations of this script:
+            for old_file in [outname, trailer]:
+                if os.access(old_file, os.F_OK):
+                    os.remove(old_file)
+            res = calstis(os.path.basename(inflat), wavecal=wavecal, outroot=outroot,
+                          trailer=trailer)
+            if res != 0:
+                raise Exception('CalSTIS returned non-zero code:  {}\n' \
+                                'See log for more details:  {}'.format(res, trailer))
+            print ('File written:  {}'.format(outname))
+        finally:
+            os.chdir(old_cwd)
+    else:  # not do_cal
+        outname = inflat
+
+        # Check if user-supplied inflat has all the recommended calibrations performed:
+        keyword_warnings = []
+        for keyword in perform:
+            if hdr0[keyword] != 'COMPLETE':
+                keyword_warnings.append(keyword)
+        if keyword_warnings:
+            warnings.warn('These calibration steps should be COMPLETE:\n{}'.format(
+                ', '.join(keyword_warnings)))
+
+    # Read in the calibrated flat data:
+    data = fits.getdata(outname, ext=1)
+    numrows, numcols = np.shape(data)
 
     # Do a line-by-line cubic spline fit to the fringe flat to remove the lamp function...
 
-    # If short-slit aperture (does not start with "52X"):
-    #    Find the row with max counts in the short-slit fringe flat
-    #    Uses central 60% of columns
-    #    Does some flux-filtering
+    with fits.open(outname) as hdulist:
+
+        aperture = hdulist[0].header['APERTURE']
+        bincols = hdulist[0].header['BINAXIS1']
+        binrows = hdulist[0].header['BINAXIS2']
+        cenwave = hdulist[0].header['CENWAVE']
+
+        # If short-slit aperture (does not start with "52X"):
+        if aperture[0:3] != "52X":
+            flatdata = hdulist[1].data
+            #    Find the row with max counts in the short-slit fringe flat
+            #    Search between the middle 10% of rows and the middle 60% of columns
+            t_row = int(0.45 * numrows)
```

Comment:
In the cl script, the corresponding lines use `nint()` (nearest integer) rather than `int()` (truncate to an int).  In most cases you can use `round()` in Python instead of `nint()` in iraf.  Here's an example where `nint` and `int` differ:

    ecl> =nint(0.2 * 1024)
    205

    int(0.2 * 1024)
    204
    round(0.2 * 1024)
    205